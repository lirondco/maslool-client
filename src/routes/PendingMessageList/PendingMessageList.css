.PendingMessageList {
  margin: 2%;
  padding: 2%;
}

.PendingMessageList h2 {
  margin-bottom: 1%;
}

hr {
  border: 1px solid;
}

.message_list {
  display: flex;
  flex-direction: column;
  margin-top: 3%;
}
