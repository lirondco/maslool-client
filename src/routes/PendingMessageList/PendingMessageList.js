import React, { Component } from 'react'
import PendingMessage from '../../components/PendingMessage/PendingMessage'
import PendingApiService from '../../services/pending-api-service'
import './PendingMessageList.css'

export default class PendingMessageList extends Component {
    state = {
        pending: null,
        error: null
    }

    loadAllPending = () => {
        this.setState({ error: 'Loading messages ...' })
        PendingApiService.getAllPending()
            .then(pending => this.setState({ pending, error: null }))
            .catch(er => this.setState({ er }))
    }

    componentDidMount = () => {
        this.loadAllPending()
    }

    handleDeleteClick = (pendingId, e) => {
        PendingApiService.deletePending(pendingId)
            .then(() => this.loadAllPending())
            .catch(er => this.setState({ er }))
    }

    renderMessages = () => {
        const { pending } = this.state
        return (
            <>
                <ul className='message_list'>
                    {pending.map(pending =>
                        <li key={pending.id}>
                                <PendingMessage pending={pending} onDeleteClick={e => this.handleDeleteClick(pending.id, e)} />
                        </li>
                    )}
                </ul>
            </>
        )
    }

    render() {
        return (
            <section className='PendingMessageList'>
                <h2>Pending Messages</h2>
                <hr />
                {(this.state.pending === null)
                    ? <h4>You have no pending messages.</h4>
                    : this.renderMessages()
                }
                <div role='alert'>
                    {this.state.error && <p className="error">{this.state.error}</p>}
                </div>
            </section>
        )
    }
}