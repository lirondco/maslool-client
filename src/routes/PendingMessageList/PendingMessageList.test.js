import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import PendingMessageList from "./PendingMessageList";

describe(`PendingMessageList component`, () => {
  it("renders the complete list", () => {
    const wrapper = shallow(<PendingMessageList />);
    wrapper.setState({
      error: null,
      pending: [
        {
          id: 9,
          message: "test message",
          user: {
            id: 1,
            username: "spongebob",
            email: "sb.squarepants@bikinibottom.net",
            join_date: "2021-03-11T06:47:06.477Z",
            admin: false,
            banned: false,
          },
        },
      ],
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
