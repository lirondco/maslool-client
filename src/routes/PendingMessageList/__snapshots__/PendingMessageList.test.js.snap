// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingMessageList component renders the complete list 1`] = `
<section
  className="PendingMessageList"
>
  <h2>
    Pending Messages
  </h2>
  <hr />
  <ul
    className="message_list"
  >
    <li
      key="9"
    >
      <PendingMessage
        onDeleteClick={[Function]}
        pending={
          Object {
            "id": 9,
            "message": "test message",
            "user": Object {
              "admin": false,
              "banned": false,
              "email": "sb.squarepants@bikinibottom.net",
              "id": 1,
              "join_date": "2021-03-11T06:47:06.477Z",
              "username": "spongebob",
            },
          }
        }
      />
    </li>
  </ul>
  <div
    role="alert"
  />
</section>
`;
